ensime_shared/ensime.py

```diff
@@ -303,7 +303,7 @@ def register_responses_handlers(self):
         self.handlers["IndexerReadyEvent"] = f_indexer
         f_indexer = lambda ci, p: self.message("analyzer_ready")
         self.handlers["AnalyzerReadyEvent"] = f_indexer
-        self.handlers["NewScalaNotesEvent"] = self.handle_new_scala_notes_event
+        self.handlers["NewScalaNotesEvent"] = self.handle_new_scala_notes_event_with_syntastic if self.vim.eval('exists("g:SyntasticRegistry")') else self.handle_new_scala_notes_event
```

Comment:
Please, all the vim commands put them in the `commands` dictionary and use `vim_eval`. That way we have all the 'vim' commands gathered in only one place. :smile: